DEVELOPING.md

```diff
@@ -1,75 +1,236 @@
+└───.github/
+    ├── CODE_OF_CONDUCT.md      # The code of conduct
+    └── workflows/              # GitHub CI workflow definition files
+        ├── pr.yml              # The CI workflow that validates PRs
+        └── main.yml            # Build & publish master branch to pre-release
+```
 
 
-Building TLC / `tlatools.jar`
-=============================
+Build & Test CLI Tools
+----------------------
+Install the following dependencies to your path:
+ * [OpenJDK](https://openjdk.org/) version 11+
+ * [Apache Ant](https://ant.apache.org/) version 1.9.8+
 
-While the Toolbox is maintained in [Eclipse](https://github.com/tlaplus/tlaplus/tree/master/general/ide), sometimes changes to TLC do not require building the entire GUI. In such cases, it can be beneficial to compile TLC on the command line. 
+Clone this repo & open a shell in its root, then run:
+```bash
+cd tlatools/org.lamport.tlatools
+ant -f customBuild.xml compile compile-test dist # Builds tla2tools.jar
+java -cp dist/tla2tools.jar tlc2.TLC test-model/pcal/Bakery.tla # Runs an example
+ant -f customBuild.xml test # Runs unit tests (optional)
+```
 
-While `Maven` is used for the Toolbox (see below), `ant` can be used for the command line. It can be installed with `brew` or `apt` with the package name `ant` (e.g. `brew install ant` and `sudo apt-get install ant`).
+The unit tests should all succeed.
+The compiled `tla2tools.jar` will be output to `tlatools/org.lamport.tlatools/dist/tla2tools.jar`, and can be used the same as any `tla2tools.jar` you download from the releases.
 
-Make sure that you're using Java 11+. If you're using a lower version you'll see compilation errors.
+Build & Test Toolbox IDE
+------------------------
+*Note: these instructions are likely to fail and must be fixed; see [this issue](https://github.com/tlaplus/tlaplus/issues/851).*
 
-You can build TLC from a fresh clone of this repository+:
+Install the following dependencies to your path:
+ * [OpenJDK](https://openjdk.org/) version 11+
+ * [Apache Ant](https://ant.apache.org/) version 1.9.8+
+ * [Apache Maven](https://maven.apache.org/) version 3.9.7+
+
+Clone this repo & open a shell in its root, then run:
 ```bash
-$ git clone https://github.com/tlaplus.git
-$ cd tlaplus/tlatools/org.lamport.tlatools
-$ ant -f customBuild.xml info clean compile compile-test dist
+mvn install -Dmaven.test.skip=true # Builds Toolbox
+mvn verify # Runs tests
 ```
 
-The `tlatools.jar` will be created in the `org.lamport.tlatools/dist` directory.
-
-Test that the build was successful by using an example model:
+The Toolbox distributables will be output in the `toolbox/org.lamport.tla.toolbox.product.product/target/products` directory.
+
+Developing with the Eclipse IDE
+-------------------------------
+
+You are free to use your own editor, but this project is largely developed in the [Eclipse](https://www.eclipse.org/) integrated development environment (IDE).
```

Comment:
One can develop the TLA+ Tools in pretty much an IDE, but Eclipse is the way to go when it comes to the Toolbox (anything else will be an uphill battle).  Thus, I suggest that this section is split into a TLA+ Tools and a Toolbox section, with a focus on the TLA+ Tools --- we want more TLA+ Tool contributors.